### PR TITLE
Add documentation for state mismatch between db and executor

### PR DESCRIPTION
**Pull request by RNHTTR**
This was discussed at length in [this comment](https://github.com/apache/airflow/pull/40468#discussion_r1670868964) to #40468

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
I lke the approach "something wrong" is good enough, it's great that this is actionable.

File: docs/apache-airflow/troubleshooting.rst [hunk 1]
```diff
@@ -20,6 +20,19 @@
 Troubleshooting
 ===============
 
+Obscure scheduling failures
```
Comment (eladkal) [inline docs/apache-airflow/troubleshooting.rst:23]: I rather we won't place this here.

Our goal should be shorting this doc not adding more stuff to it.

I rather we explain the mechanisem of scheduler-executor dynamic in the relevant section of the docs and then explain the limits/edge cases.

File: docs/apache-airflow/troubleshooting.rst [hunk 2]
```diff
@@ -20,6 +20,19 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+State mismatch between Airflow database and executor
+----------------------------------------------------
+
+This indicates that when the scheduler queried the Airflow database, it observed that the task instance had one status according to the Airflow metadata database,
+but a different status according to the executor. A common example is when the query returned to the scheduler, the task instance was in the ``queued`` status,
```
Comment (eladkal) [inline docs/apache-airflow/troubleshooting.rst:30]: What query?

File: docs/apache-airflow/troubleshooting.rst [hunk 3]
```diff
@@ -20,6 +20,19 @@
...
+but the status according to the executor was ``running``.
+
+This mismatch must have persisted for multiple attempts. When this happens, Airflow will not attempt to queue the task. It's possible that something has gone wrong
```
Comment (eladkal) [inline docs/apache-airflow/troubleshooting.rst:33]: Attempts=airflow retries?

Comment (RNHTTR) [reply docs/apache-airflow/troubleshooting.rst:33]: no -- good call that this needs to be clarified

File: docs/apache-airflow/troubleshooting.rst (hunk 2 above)
Comment (RNHTTR) [reply docs/apache-airflow/troubleshooting.rst:30]: i will clarify which query.

File: docs/apache-airflow/troubleshooting.rst (hunk 1 above)
Comment (RNHTTR) [reply docs/apache-airflow/troubleshooting.rst:23]: This particular log message (`"could not queue task %s (still running after %d attempts)"`) is essentially useless. The troubleshooting page adds context that we can't fit into a log message. It'd be difficult to provide this context in a way users can find in the normal documentation.

Comment (eladkal) [reply docs/apache-airflow/troubleshooting.rst:23]: But you need to consider that people land in the doc without hitting the message you are refering to.

I wonder if maybe the message should refer to github discussion arround the problem. I am not a fan of troubleshooting pages. If this is a bug/limitation/problem we don't have a good solution for then it's an open issue.

**Review by hussein-awala — APPROVED** [review]

**Review by pierrejeambrun — APPROVED** [review]

**Pull request closed without merging**